### PR TITLE
remove 'initial user' workflow

### PR DESCRIPTION
**Pull request by mchalski**
this PR removes any and all notion of an 'initial user' from the system. from now on,
all users must be created a priori via the CLI, and can log in only with their full
credentials (Basic Auth -> JWT with the 'mender.*' scope).

changes include:
- removed POST /users/initial
- removed the intial user logic from POST /login
- removed intial user logic in POST /auth/verify (forgot to create a task for this, handling it here)

note that this *will* break stuff:
- integration tests must be adjusted:
https://github.com/mendersoftware/integration/blob/fda7608e107749b2bb50ec09eb01f5a89366876d/tests/MenderAPI/authentication.py#L25
- the UI's 'create user' screen will be broken (until https://tracker.mender.io/browse/MEN-1033 is done)

EDIT: the required integration test update is here, merge it first:
https://github.com/mendersoftware/integration/pull/232

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
backend has many of these errors:
```

12:16:11 Aborting.
12:16:11 INFO:root:getting all devices from :https://172.18.0.19/api/management/v1/admission/devices
12:16:11 INFO:root:fail to get devices (payload: <html>
12:16:11 <head><title>401 Authorization Required</title></head>
12:16:11 <body bgcolor="white">
12:16:11 <center><h1>401 Authorization Required</h1></center>
12:16:11 <hr><center>openresty/1.11.2.3</center>
12:16:11 </body>
12:16:11 </html>
```

**Comment by mchalski** [comment]
@GregorioDiStefano this PR needs to be tested with this integration test update:
https://github.com/mendersoftware/integration/pull/232

can you do the jenkins magic to get a green build on both PRs?

**Comment by GregorioDiStefano** [comment]
this build is running now, and it's broken

File: docs/management_api.yml [hunk 1]
```diff
@@ -68,52 +58,6 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
-  /users/inital:
```
Comment (bboozzoo) [inline docs/management_api.yml]: Looks like we'll need to increment API version increment this time. @maciejmrowiec comments/objections?

**Comment by GregorioDiStefano** [comment]
@mchalski this test run will fail, but i'll soon merge a fix for the spurious fails i mentioned in the standup, once that's it in, we can try re-running this

File: docs/management_api.yml (hunk 1 above)
Comment (mchalski) [inline docs/management_api.yml]: @maciejmrowiec ping

@bboozzoo you're right. how should we bump this version?

intuitively, it's a major change - api is definitely broken, so we should move to `2.0.0 = v2`.

OTOH the only client here is the gui, and it's distributed with the backend...so we could get away with a minor update.

OTOH yet - maybe some users have integrated e.g. their own scripts which use the mgmt api, and we might break them. so I'd lean tworads a major update. thoughts?

Comment (bboozzoo) [inline docs/management_api.yml]: not my call, @maciejmrowiec ?

Comment (kjaskiewiczz) [inline docs/management_api.yml]: imo, we don't need to bump API version
do we know about any clients that did integration with this part of API?

File: .travis.yml [hunk 2]
```diff
@@ -107,27 +105,6 @@ script:
     # Test packages pararell (xargs -P)
     - go list ./... | grep -v vendor | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {}
 
-    # Insert build information into compiled binary using ldflags
```
Comment (maciejmrowiec) [inline .travis.yml]: If we remove building binary and subsequent build of container, there will be nothing to be pushed to docker hub in git push.

Building container also tests docker file correctness on it's own.

Comment (mchalski) [inline .travis.yml]: true, bringing it back

**Pull request merged**